dart/dynamics/ReferentialSkeleton.hpp

```diff
@@ -460,6 +479,13 @@ class ReferentialSkeleton : public MetaSkeleton
   /// Name of this ReferentialSkeleton
   std::string mName;
 
+  /// Skeletons that this ReferentialSkeleton is referencing to.
+  std::unordered_set<const Skeleton*> mSkeletons;
+
+  /// Mutexes of the skeletons. The mutexes are sorted in order of memory
+  /// addresses.
+  std::set<std::mutex*> mSkeletonMutexes;
```

Comment:
This seems like an implementation detail. Could we just return a `common::MutexReference` here?

Reply:
You mean you suggest removing `getSkeletonMutexes()`? If so, it's removed.